chain/chunks/src/lib.rs

```diff
@@ -677,12 +704,37 @@ impl ShardsManager {
 
         let can_reconstruct = entry.parts.len() >= self.runtime_adapter.num_data_parts();
 
-        if have_all_parts {
+        let chunk_producer = self.runtime_adapter.get_chunk_producer(
+            &self.runtime_adapter.get_epoch_id_from_prev_block(&prev_block_hash)?,
```

Comment:
We already retrieve current epoch above (line 604 before your change 631 with your change), just don't store it. Store it there and don't re-retrieve here.

Reply:
++